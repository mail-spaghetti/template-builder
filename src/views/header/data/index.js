export const ButtonData = [
  {
    text: "Undo",
  },
  {
    text: "Save",
  },
  {
    text: "History",
  },
  {
    text: "Preview",
  },
];
