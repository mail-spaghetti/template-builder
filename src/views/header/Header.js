import React from "react";
import Button from "../../components/atoms/Button";
import { ButtonData } from "./data";

import Logo from "../../utils/icons/Logo";

const Header = (props) => (
  <div className="header">
    <div className="header__logo">
      <Logo />
    </div>
    <div className="header__cta">
      {ButtonData.map((d, idx) => (
        <Button key={idx} variant="secondary" {...d} />
      ))}
    </div>
    <div className="header__auth">
      <span className="header__caption">User Name</span>
      <span>
        <figure className="header__shape">
          {props.img ? (
            <img className="header__img" alt="Authenticated user" />
          ) : null}
        </figure>
      </span>
    </div>
  </div>
);

export default Header;
