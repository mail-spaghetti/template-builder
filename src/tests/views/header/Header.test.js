import React from 'react';

import { shallow } from "enzyme";

import { Header } from '../../../pages/header/Header';

test("Should render Header correctly", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
})