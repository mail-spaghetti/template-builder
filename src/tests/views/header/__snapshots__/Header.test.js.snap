// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div
  className="header"
>
  <div
    className="header__logo"
  >
    <Logo />
  </div>
  <div
    className="header__cta"
  >
    <Button
      icon="Replay"
      key="0"
      text="Undo"
      variant="secondary"
    />
    <Button
      icon="Floppy"
      key="1"
      text="Save"
      variant="secondary"
    />
    <Button
      icon="History"
      key="2"
      text="History"
      variant="secondary"
    />
    <Button
      icon="Preview"
      key="3"
      text="Preview"
      variant="secondary"
    />
  </div>
  <div
    className="header__auth"
  >
    <span
      className="header__caption"
    >
      User Name
    </span>
    <span>
      <figure
        className="header__shape"
      />
    </span>
  </div>
</div>
`;
